docs/docs/how-to/testing/testing-components-with-graphql.md

```diff
@@ -2,80 +2,34 @@
-simplest thing to do is to test the individual page components with mock data,
-rather than trying to test a full page. However, if you do want to test the full
-page you'll need to provide the equivalent data to the component. Luckily
-there's a way to get the data you need.
-
-First you should make sure you have read
-[the unit testing guide](/docs/how-to/testing/unit-testing/) and set up your project as
-described. This guide is based on the same blog starter project. You will be
-writing a snapshot test for the index page.
-
-As Jest doesn't run or compile away your GraphQL queries you need to mock the
-`graphql` function to stop it throwing an error. If you set your project up with
-a mock for `gatsby` as described in the unit testing guide then this is already
-done.
+If you try to run unit tests on components that use GraphQL queries, you will discover that you have no data. Jest can't run your queries, so if you are testing components that rely on GraphQL data, you will need to provide the data yourself. This is a good thing, as otherwise your tests could break if your data changes, and in the case of remote data sources it would need network access to run tests.
 
-## Testing page queries
+In general it is best practice to test the smallest components possible, so the simplest thing to do is to test the individual page components with mock data, rather than trying to test a full page. However, if you do want to test the full page you'll need to provide the equivalent data to the component. Luckily there's a way to get the data you need.
 
-As this is testing a page component you will need to put your tests in another
-folder so that Gatsby doesn't try to turn the tests into pages.
+First you should make sure you have read [the unit testing guide](/docs/how-to/testing/unit-testing/) and set up your project as described. This guide is based on the same blog starter project. You will be writing a snapshot test for the index page.
 
-```js:title=src/pages/__tests__/index.js
-import React from "react"
-import renderer from "react-test-renderer"
-import Index from "../index"
+As Jest doesn't run or compile away your GraphQL queries you need to mock the `graphql` function to stop it throwing an error. If you set your project up with a mock for `gatsby` as described in the unit testing guide then this is already done.
 
-describe("Index", () =>
-  it("renders correctly", () => {
-    const tree = renderer.create(<Index />).toJSON()
-    expect(tree).toMatchSnapshot()
-  }))
-```
+## Testing page queries
 
-If you run this test you will get an error, as the StaticQuery in the `Layout` component is not mocked. You can fix this by mocking it, like so:
+As this is testing a page component you will need to put your tests in another folder so that Gatsby doesn't try to turn the tests into pages.
 
-```js:title=src/__tests__/index.js
+```js:title=src/pages/__tests__/index.js
```

Comment:
````suggestion
```tsx:title=src/pages/__tests__/index.js
````

Reply:
because of component at `renderer.create(<PageTwo />)` should this file not be a `index.jsx`?